packages/messages/src/browser/notification-center-component.tsx

```diff
@@ -0,0 +1,146 @@
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+import { DisposableCollection } from '@theia/core';
+import { NotificationManager, Notification } from './notifications-manager';
+
+const PerfectScrollbar = require('react-perfect-scrollbar');
+
+export interface NotificationCenterComponentProps {
+    readonly manager: NotificationManager;
+}
+
+interface NotificationCenterComponentState extends NotificationManager.UpdateEvent { }
+
+export class NotificationCenterComponent extends React.Component<NotificationCenterComponentProps, NotificationCenterComponentState> {
+
+    constructor(props: NotificationCenterComponentProps) {
+        super(props);
+        this.state = {
+            notifications: [],
+            open: false
+        };
+    }
+
+    protected readonly toDisposeOnUnmount = new DisposableCollection();
+
+    async componentDidMount() {
+        this.toDisposeOnUnmount.push(
+            this.props.manager.onUpdate(event => {
+                this.setState({
+                    notifications: event.notifications.filter(n => n.location === 'notification'),
+                    open: event.open
+                });
+            })
+        );
+    }
+    componentWillUnmount() {
+        this.toDisposeOnUnmount.dispose();
+    }
+
+    render() {
+        const empty = this.state.notifications.length === 0;
+        const showHeader = empty ? { display: 'flex' } : {};
+        const title = empty ? 'NO NOTIFICATIONS' : 'NOTIFICATIONS';
+        return (
+            <div className={`theia-notification-center ${this.state.open ? 'open' : 'closed'}`}>
+                <div className='theia-notification-center-header' style={showHeader}>
+                    <div className='theia-notification-center-header-title'>{title}</div>
+                    <div className='theia-notification-center-header-actions'>
+                        <ul className='theia-notification-actions'>
+                            <li className='collapse' title='Hide' onClick={this.onHide.bind(this)} />
+                            <li className='clear' title='Clear All' onClick={this.onClearAll.bind(this)} />
+                        </ul>
+                    </div>
+                </div>
+                <PerfectScrollbar className='theia-notification-list-scroll-container'>
+                    <div className='theia-notification-list'>
+                        {this.state.notifications.map(notification => this.renderNotification(notification))}
+                    </div>
+                </PerfectScrollbar>
+            </div>
+        );
+    }
+
+    protected onHide() {
```

Comment:
Should we include the `return-type` for these methods?
https://github.com/theia-ide/theia/wiki/Coding-Guidelines#explicit-return-type

Reply:
@vince-fugnitto I wonder is there a tslint rule for it?

Reply:
https://palantir.github.io/tslint/rules/typedef/ @vince-fugnitto Could you try to make a PR with it as a warning?

Reply:
fixed